generator/generator.md

```diff
@@ -0,0 +1,246 @@
+
+> <span id="par">**Note:**</span> The `parentheses` field is optional and allows you to add priority to the conditional logic by using opening and closing parentheses. This is useful for specifying the order in which conditions should be evaluated. For example, you can use parentheses to create complex conditions like `conditional1 || (conditional2 && conditional3)`, where `conditional2 && conditional3` is evaluated first due to the parentheses.
+
+> <span id="rel">**Note:**</span> `${relativePath}` in `path` is used to obtain the relative path within the same group, facilitating the reference to data points that share a common parent or group with the current data point being processed in the TypeScript code for generating conditionals.
+
+### Conditionals TypeScript
+
+```typescript
+// customConditionals.tsx
+
+// In this example, the age of the person is equal or more than 16 and the person have a driving license 
+export const siPermisConduireConditional: Conditional = (interview, path) => {
+    const relativePath = path.substring(0, path.lastIndexOf('.')); // Remove the last key from the path
+    return checkConditionals({
+        interview,
+        conditionals: [
+            {
+                path: `${relativePath}.age`,
+                comparisonOperator: '>=',
+                value: 16,
+            },
+            {
+                logicalOperator: '&&',
+                path: `${relativePath}.permisConduire`,
+                comparisonOperator: '===',
+                value: 'oui',
+            }
+        ]
+    });
+};
+```
+
+## Generate Choices
+
+Choices in your survey define the available options in ­­`RadioInput` or `InputCheckbox` for respondents. The table below outlines the fields in Choices tab used to define choices, along with an example and the expected output in a ­`choices.tsx­` file.
+
+### Choices Fields
+
+| Field                        | Description                                            | Example        | Type             |
+|------------------------------|--------------------------------------------------------|----------------|------------------|
+| choicesName                  | Name for the choices group                             | ouiNonChoices  | string           |
+| value                        | Unique value for the choice                            | oui            | string or number |
+| fr                           | French label for the choice                            | Oui            | string or number |
+| en                           | English label for the choice                           | Yes            | string or number |
+| [spreadChoicesName](#spread) | Spreading another choicesName (optional)               |                | string?          |
```

Comment:
I'm not sure I understand the syntax here: `[spreadChoicesName](#spread)`

Reply:
This allows it to create a link with the note, to make it easier for the user when reading on GitHub.